kyo-direct/shared/src/main/scala/kyo/Direct.scala

```diff
@@ -4,40 +4,100 @@ import cps.CpsMonad
 import cps.CpsMonadContext
 import cps.async
 import directInternal.*
+import kyo.Ansi.*
 import scala.annotation.tailrec
 import scala.quoted.*
 
-/** Defers the execution of a block of code, allowing the use of `await` within it.
+/** Defers the execution of a block of code, allowing the use of `.now` and `.later` for effect handling.
   *
-  * This macro transforms the given block of code to work with effectful computations, enabling the use of `await` to handle various types
-  * of effects. The `defer` block is desugared into regular monadic composition using `map`, making it equivalent to writing out the monadic
-  * code explicitly.
+  * This macro transforms the given block of code to work with effectful computations. Effects can be:
+  *   - Sequenced immediately using `.now` when you need their results
+  *   - Preserved using `.later` (advanced API) for more controlled effect composition
+  *
+  * The `.later` operation is an advanced feature that gives more control over effect sequencing, but requires a deeper understanding of
+  * effect composition.
+  *
+  * The `defer` block is desugared into regular monadic composition using Kyo's monadic bind (`map`).
   *
   * @tparam A
   *   The type of the value returned by the deferred block
   * @param f
   *   The block of code to be deferred
   * @return
-  *   A value of type `A < S`, where `S` represents the combined effects of all `await` calls
+  *   A value of type `A < S`, where `S` represents the combined effects of all operations
   */
 transparent inline def defer[A](inline f: A) = ${ impl[A]('f) }
 
-/** Awaits the result of an effectful computation.
-  *
-  * This function can only be used within a `defer` block. It suspends the execution of the current block until the result of the effectful
-  * computation is available. In the desugared monadic composition, `await` calls are transformed into appropriate `map` operations.
-  *
-  * @tparam A
-  *   The type of the value being awaited
-  * @tparam S
-  *   The type of the effect (e.g., IO, Abort, Choice, etc.)
-  * @param v
-  *   The effectful computation to await
-  * @return
-  *   The result of type A, once the computation completes
-  */
-inline def await[A, S](v: A < S): A =
-    compiletime.error("`await` must be used within a `defer` block")
+extension [A, S](inline self: A < S)
+
+    /** Sequences an effect immediately, making its result available for use.
+      *
+      * Must be used within a `defer` block. This operation tells the direct syntax to execute the effect at this point in the computation,
+      * allowing you to use its result in subsequent operations.
+      *
+      * @return
+      *   The unwrapped value of type `A` from the effect
+      * @throws RuntimeException
+      *   if used outside a `defer` block
+      */
+    inline def now: A = ${ nowImpl('self) }
```

Comment:
I've made these a macro so it's possible to format the error message with `Ansi`. `compiletime.error` doesn't support that.